auth/jwt_controller_test.go

```diff
@@ -95,6 +96,32 @@ func TestJWTController(t *testing.T) {
 		assert.Equal(t, map[string]string{"error": server.Lang.GetDefault().Get("auth.invalid-credentials")}, respBody)
 	})
 
+	t.Run("Login_invalid_username", func(t *testing.T) {
+		server, user := prepareAuthenticatorTest(t)
+		server.Config().Set("auth.jwt.secret", "secret")
+
+		mockUserService := &MockUserService[TestUser]{err: fmt.Errorf("test errors: %w", gorm.ErrRecordNotFound)}
+		controller := NewJWTController(mockUserService, "Password")
+		server.RegisterRoutes(func(_ *goyave.Server, router *goyave.Router) {
+			router.Controller(controller)
+		})
+
+		data := map[string]any{
+			"username": "wrong username",
+			"password": user.Password,
+		}
+		body, err := json.Marshal(data)
+		require.NoError(t, err)
+		request := httptest.NewRequest(http.MethodPost, "/login", bytes.NewReader(body))
+		request.Header.Set("Content-Type", "application/json")
+		resp := server.TestRequest(request)
+		assert.Equal(t, http.StatusUnauthorized, resp.StatusCode)
+		respBody, err := testutil.ReadJSONBody[map[string]string](resp.Body)
+		assert.NoError(t, resp.Body.Close())
+		require.NoError(t, err)
```

Comment:
I would invert these two lines

```suggestion
		require.NoError(t, err)
		assert.NoError(t, resp.Body.Close())
```

Because if an error occurs, resp will be nil, so resp.Body.Close() would panic

Reply:
We are testing the error from `testutil.ReadJSONBody` here. The `resp` won't be `nil` at this point.